src/plugins/hetero/src/plugin.cpp

```diff
@@ -329,5 +331,10 @@ ov::SoPtr<ov::IRemoteContext> ov::hetero::Plugin::create_context(const ov::AnyMa
 }
 
 ov::SoPtr<ov::IRemoteContext> ov::hetero::Plugin::get_default_context(const ov::AnyMap& remote_properties) const {
-    OPENVINO_NOT_IMPLEMENTED;
+    std::map<std::string, ov::SoPtr<ov::IRemoteContext>> contexts_map;
+    OPENVINO_ASSERT(execution_devices.size() >= 1, "[HETERO] No execution devices found.");
+    for (const auto& device : execution_devices) {
+        contexts_map.insert({device, get_core()->get_default_context(device)});
+    }
+    return std::make_shared<ov::hetero::RemoteContext>(contexts_map);
```

Comment:
actually I'm concerned that this impl is somehow not thread-safe, even though it's not a common case in hetero to compile multiple models at same time...

Reply:
refactored the code, constructed the remote context before compiling the model.